src/include/settings/settings_type.h

```diff
@@ -0,0 +1,29 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// settings_type.h
+//
+// Identification: src/include/settings/settings_type.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#pragma once
+
+#define __SETTING_ENUM__
+#include "settings/settings_macro.h"
+
+namespace peloton {
+namespace settings {
+
+enum class SettingsId {
+  #include "settings/settings.h"
+};
+
+}  // namespace settings
+}  // namespace peloton
+
+#undef __SETTING_ENUM__
```

Comment:
1. Should we rename this to `setting_id.h`
2. Can you rename `SettingsId` -> `SettingId`.
3. Can you move the #define/#undefine into the enumeration to make clear the scope? Something like:
```
enum class SettingId {
  #define __SETTING_ENUM__
  #include "settings/settings.h"
  #undef __SETTING_ENUM__
};
```